### PR TITLE
feat(StoneDB 8.0): fix stonedb 8.0 compile failed (#606)

### LINKED ISSUE #606: feature(StoneDB 8.0):  compile the stonedb 8.0  failed
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

 compile the stonedb 8.0  failed, because MySQL 8.0's boost patches conflict with boost code
 MySQL 8.0 add some code in  boost patches, we can see :https://github.com/mysql/mysql-server/commit/e11fd7d08469185562774b4d0b7e1a6bd47bd38c#
so we need to set the compilation order and give priority to the boost code in patches

**Describe alternatives you've considered**

modify cmake，give priority to the boost code in patches

**Additional context**

### PR DESCRIPTION
**Pull request by DandreChen**
## Summary about this PR

Issue Number: close #606 

[summary]
1 during compilation, errors will be reported that the boost function or variable is not defined 
2 modify the cmake file that uses the boost module, so that the boost code in patches can be used first

## Tests Check List

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

## Changelog

- [ ] New Feature
- [x] Bug Fix
- [ ] Improvement
- [ ] Performance Improvement
- [ ] Build/Testing/CI/CD
- [ ] Documentation
- [ ] Not for changelog (changelog entry is not required)

## Documentation

- [x] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features

### CODE REVIEW

**Review by lujiashun — APPROVED** [review]
LGTM

File: client/dump/CMakeLists.txt [hunk 1]
```diff
@@ -28,7 +28,7 @@ IF(USING_LOCAL_BOOST)
     -DBOOST_SYSTEM_NO_DEPRECATED)
 ENDIF()
 
-INCLUDE_DIRECTORIES(SYSTEM ${BOOST_PATCHES_DIR} ${BOOST_INCLUDE_DIR})
+INCLUDE_DIRECTORIES(SYSTEM ${BOOST_PATCHES_DIR} ${BOOST_INCLUDE_DIR}/include)
```
Comment (RingsC) [inline client/dump/CMakeLists.txt:31]: Pls, make the sure the precise location of `${BOOST_INCLUDE_DIR}`. `${BOOST_INCLUDE_DIR}`, is it point to `/path/to/boost/` or `/path/to/boost/include`?

File: cmake/fastcov.cmake [hunk 2]
```diff
@@ -115,7 +115,7 @@ ENDFOREACH()
 SET(FASTCOV_EXCLUDE_LIST "--exclude")
 FOREACH(FASTCOV_EXCLUDE
     "/usr/include"
-    "${BOOST_INCLUDE_DIR}"
+    "${BOOST_INCLUDE_DIR}/include"
```
Comment (RingsC) [inline cmake/fastcov.cmake:118]: same as above.

**Review by RingsC — CHANGES_REQUESTED** [review]

File: client/dump/CMakeLists.txt (hunk 1 above)
Comment (RingsC) [reply client/dump/CMakeLists.txt:31]: ${BOOST_INCLUDE_DIR} is：`/usr/locak/stonedb_boost` on my instance.

Comment (RingsC) [reply client/dump/CMakeLists.txt:31]: > Pls, make the sure the precise location of `${BOOST_INCLUDE_DIR}`. `${BOOST_INCLUDE_DIR}`, is it point to `/path/to/boost/` or `/path/to/boost/include`?

${BOOST_INCLUDE_DIR} is：/usr/locak/stonedb_boost on my instance.

**Review by RingsC — APPROVED** [review]
LGTM

**Pull request merged**